src/presign/record.rs

```diff
@@ -121,7 +121,8 @@ impl PresignRecord {
     /// Compute the x-projection of the randomly-selected point `R` from the
     /// [`PresignRecord`].
     pub(crate) fn x_projection(&self) -> Result<Scalar> {
-        let x_projection = self.R.0.to_affine().x();
+        //let x_projection = self.R.0.to_affine().x();
+        let x_projection = self.R.get_zero().to_affine().x();
```

Comment:
todo: to me, getting the x-coordinate of a `CurvePoint` is a really natural method, so I'd rather have a method `x` or `x_projection` on that type. 

I think what I'm realizing looking through this is that I don't like the pattern of converting a `CurvePoint` to a `ProjectivePoint` and then to an `AffinePoint`. Our `CurvePoint` abstraction should handle all of those transformations so when we use it in the code, we can focus on the actual, relevant operation rather than the internal type transformations.

Also, please remove commented-out code.